bugrap/src/main/java/org/vaadin/bugrap/BaseModel.java

```diff
@@ -1,12 +1,33 @@
 package org.vaadin.bugrap;
 
 import org.vaadin.bugrap.domain.BugrapRepository;
+import org.vaadin.bugrap.domain.entities.Reporter;
+
+import com.vaadin.navigator.Navigator;
 
 public class BaseModel {
 
+	public static final String NAV_LOGIN 	= "login";
+	public static final String NAV_REPORT	= "report";
+	
 	private static final BugrapRepository repository = new BugrapRepository(DatabaseHelper.DB_LOCATION);
+	private Navigator navigator;
```

Comment:
There is a way of getting the navigator associated with the current UI, so you shouldn't need to keep passing its reference around.